include/xgboost/data.h

```diff
@@ -270,6 +285,11 @@ class SparsePage {
             static_cast<Inst::index_type>(size)};
   }
 
+  HostSparsePageView GetView() const {
```

Comment:
Is there a way to fix operator[] instead of creating a whole new class?

Reply:
@RAMitchell If you want to call `operator[]` without first calling `HostVector` twice, then I don't think so.